tokenizers/src/tokenizer/added_vocabulary.rs

```diff
@@ -90,6 +91,37 @@ impl std::cmp::Eq for AddedToken {}
 
 type MatchingSet = (AhoCorasick, Vec<u32>);
 
+lazy_static! {
+    static ref STARTS_WITH_WORD: Regex = Regex::new(r"^\w").unwrap();
+    static ref ENDS_WITH_WORD: Regex = Regex::new(r"\w$").unwrap();
+    static ref RIGHTMOST_SPACE_AT_START: Regex = Regex::new(r"^\s*").unwrap();
+    static ref LEFTMOST_SPACE_AT_END: Regex = Regex::new(r"\s*$").unwrap();
+}
+
+fn ends_with_word(sentence: &str) -> bool {
+    ENDS_WITH_WORD.is_match(sentence)
+}
+
+fn starts_with_word(sentence: &str) -> bool {
+    STARTS_WITH_WORD.is_match(sentence)
+}
+
+fn space_leftmost_at_end(sentence: &str) -> usize {
+    if let Some(match_) = LEFTMOST_SPACE_AT_END.find(sentence) {
+        match_.start()
+    } else {
+        // This should never happen since the Regex should match all the time
+        sentence.len()
+    }
```

Comment:
if it is `never happen`, can we just call `LEFTMOST_SPACE_AT_END.find(sentence).expect("LEFTMOST_SPACE_AT_END should always match but didnt match int this sentence: {sentence}").`

Reply:
`.unwrap()` is something I tend to avoid within logic code at all if possible, since having the program panic, is never a great user experience.

My answer : https://github.com/huggingface/tokenizers/pull/919#discussion_r813862919 applies too. I think returning `Option` would be preferrable to panicking, but the result would be the same IMO, if the regex doesn't match, I just don't capture anything.

Reply:
oh I see 🙂